app/controllers/question_controller.rb

```diff
@@ -2,8 +2,8 @@ class QuestionsController < ApplicationController
 	include ActionView::Helpers::TextHelper
 
 	before_action :authenticate_user!, except: [:index, :show]
-	before_action :set_question, only: [:show, :edit, :update, :destroy]
-
+	before_action :question # only: [:show, :edit, :update, :destroy]
+	before_action :user
```

Comment:
Since commenting and answering would occur on the show page, and possibly even the index page if we get fancy, I think we should identify the user, no?